include/zephyr/arch/riscv/pmp.h

```diff
@@ -11,6 +11,33 @@
 
 #ifdef CONFIG_RISCV_PMP
 
+#ifdef CONFIG_MEM_ATTR
```

Comment:
no guard here please - "[Function declarations that are available only when the option is enabled should be provided unconditionally](https://docs.zephyrproject.org/latest/develop/api/design_guidelines.html#:~:text=Function%20declarations%20that%20are%20available%20only%20when%20the%20option%20is%20enabled%20should%20be%20provided%20unconditionally)."